src/tool/xmeta/cpp/models/namespace_model.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <string>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "base_model.h"
+#include "class_model.h"
+#include "struct_model.h"
+
+
+namespace xlang
+{
+    namespace xmeta
+    {
+        struct namespace_body_model;
+        class namespace_model;
+
+        typedef std::vector<std::shared_ptr<namespace_model>> namespace_vector;
```

Comment:
Is typedef really necessary here? How is it different from the other structs?

Reply:
+1. type aliases hide the concepts (a vector of models), harming legibility.